src/catalog/catalog_set.cpp

```diff
@@ -169,7 +169,6 @@ void CatalogSet::alterEntry(Transaction* transaction, const binder::BoundAlterIn
 
 CatalogEntrySet CatalogSet::getEntries(Transaction* transaction) {
     CatalogEntrySet result;
-    std::lock_guard lck{mtx};
```

Comment:
@ray6080 had to delete this due to deadlock. Wondering if this can stay deleted? since we traverse version chains for the entries the tranactional guarantee should be fine. Just afraid of any undefined data behaviour when another connection emplaces a new entry.